### ISSUE TITLE
Allow parameter references in test expectations

### ISSUE THREAD

**Issue by guillett** [openfisca/openfisca-core#854]
We have to write this because parameter references are not understood.

Here is an exemple.
```yaml
- name: Personne seule perçoit 500 €
  period: 2019-02
  absolute_error_margin: 0.005
  input:
    paris_logement_psol_base_ressources: 500
  output:
    paris_solidarite_pa_base_ressources: 868.2 # prestations.minima_sociaux.aspa.montant_annuel_seul / 12
    paris_solidarite_pa_montant: 111.80 # plafond - prestations.minima_sociaux.aspa.montant_annuel_seul / 12
```
This is not ideal because it create dependencies with minor or patch versions in OpenFisca-France (where simple parameter changes occur).

Extract from [openfisca-paris](https://github.com/betagouv/openfisca-paris/blob/master/tests/unittests/personnes_agees/psol.yaml#L9-L16)

**Comment by Morendil** [comment]
The title of this PR seems like the description of a solution. I'm not sure I completely understand what problem would be solved.

Why would you need to change this test when changes take place in France? Parameter changes are usually the result of the law changing, and parameter values are relative to a date, so I wouldn't expect this test to change unless we change its reference period.

**Comment by guillett** [comment]
Thanks @Morendil for your quick reply. I'm sorry I bumped this issue here to lower my cognitive load but it definitely needs more information.

TL;DR: Extension with tests can't rely as much as I would like on the current SEMVER implementation.

That is why, as a co-maintainer of OpenFisca-Paris extension, I would like to test formulas at a _higher_ level then _low_ number level.

For instance, I would like to be able to write that sort of tests:
```yaml
- name: Personne seule perçoit 500 €
  period: 2019-02
  absolute_error_margin: 0.005
  input:
    paris_logement_psol_base_ressources: 500
  output:
    paris_solidarite_pa_base_ressources: prestations.minima_sociaux.aspa.montant_annuel_seul / 12
    paris_solidarite_pa_montant: paris.personnes_agees.psol.plafond.personne_seule - prestations.minima_sociaux.aspa.montant_annuel_seul / 12
```

As of now, I can't so, [we have to write](https://github.com/betagouv/openfisca-paris/blob/master/tests/unittests/personnes_agees/psol.yaml#L9-L16) tests with *explicit* numeric values:
```yaml
- name: Personne seule perçoit 500 €
  period: 2019-02
  absolute_error_margin: 0.005
  input:
    paris_logement_psol_base_ressources: 500
  output:
    paris_solidarite_pa_base_ressources: 868.2 # prestations.minima_sociaux.aspa.montant_annuel_seul / 12
    paris_solidarite_pa_montant: 111.80 # plafond - prestations.minima_sociaux.aspa.montant_annuel_seul / 12
```

Providing explicit numeric values extremely limit the compatible version range. In that specific example, that test (as it is written here) passes with OpenFisca France  [34.8.0 - #1280](https://github.com/openfisca/openfisca-france/pull/1280) but now fails with [34.7.3 - #1276](https://github.com/openfisca/openfisca-france/pull/1276).

=> Extension tests are potentially impacted by MINOR changes.

A similar situation could have occurred between [34.2.0 - #1255](https://github.com/openfisca/openfisca-france/pull/1255) and [34.2.1 -1257](https://github.com/openfisca/openfisca-france/pull/1257).

=> Extension tests are potentially impacted by PATCH changes.

cc @charlottelecuit @ManonLger

**Comment by Morendil** [comment]
I still don't understand your reference to Semver properties, we should perhaps discuss this IRL.

You seem to be looking for something like [property-based testing](https://hypothesis.readthedocs.io/en/latest/) for formulas, asserting that certain relationships hold between outputs, irrespective of (some) variation in the inputs.

Similarly to the discussion in [another issue](https://github.com/openfisca/openfisca-core/issues/853#issuecomment-475557994), implementing additional expression languages in OpenFisca, whose main job is to *be* an expression language, might not lead to a maintainable solution.

It would probably be more effective to leverage Hypothesis or even plain old Pytest and test these sort of things at the unit test level. What we need is to be able to call a function directly and pass it some values and parameters.

I've already noticed that there is lots of friction that seems to hold us back from doing that; my first recommend course of action would be to look into that friction and reduce it, rather than pile up new features of the YAML test runners, which will grow harder to maintain.

**Comment by Morendil** [comment]
@guillett OK to close this? If not, how can we make progress?

**Comment by Morendil** [comment]
Closing for lack of a plan. Feel free to reopen with comments that could suggest action.
